generic-tabs/GenericTabs.js

```diff
@@ -100,4 +101,22 @@ export class GenericTabs extends SelectedMixin(BatchingElement) {
       }
     });
   }
+
+  _preprocess() {
+    // Check if slots were setup in light DOM
+    // If not, assume heading structure and preprocess light DOM
+    if (!this.querySelector('[slot]')) {
+      const unformattedEls = [...this.children];
+      unformattedEls.forEach(node => {
+        if (node instanceof HTMLHeadingElement) {
+          const button = document.createElement('button');
+          button.textContent = node.textContent;
+          button.setAttribute('slot', 'tab');
+          node.parentNode.replaceChild(button, node);
```

Comment:
Ah, hmm... now that I am looking at the code... Do we want to be turning a heading into a button in the light DOM like that though? Seems like a thing shadow dom intends to solve.  what does it gain from this? tabindex, I guess, but it will be incorrect by default anyway (we need to set it to -1 and manage roving tab index) and role, I guess, but that will be wrong too (we change it to tab).  Why change it to rather than just set the slot tab?

Reply:
> what does it gain from this?

You gain a lot of click, focus, and keyboard behaviors right out of the box with a `button`. 

@jonathantneal's approach gets around the replacement and adds tabindex, but the last posts said we don't know if it has all the same accessibility features. https://github.com/thepassle/generic-components/pull/42#issuecomment-742649351

Reply:
Most of the things you gain are wrong and we are replacing them anyways, I think. @jonathantneal's are effectively from the ARIA practices pattern so they should totally suffice.  These are what mine do too. I think that's part of what started this - you get a lot 'for free' if they aren't tabs, but then they are actively unhelpful things that you get because they are misleading.. clicking them won't do anything, the relationships don't exist, etc.  Once we upgrade, I don't think they need to be a button anymore, the role will change, the focus behavior changes, etc.  You have to do all of the same a11y related stuff either way.  Idk, maybe @LJWatson has thoughts on whether there is something to be gained by hot swapping the light DOM into buttons.  I'd personally prefer to avoid it if possible but maybe others disagree...?

Reply:
Okay. @jonathantneal's idea is rolled in now.

Reply:
I mean..lgtm. tests pass, I haven't checked it out and tried it but yay!